### DISCUSSION TITLE
May have found a way to add a new Slingbox to your Directory - but need the FinderID

### DISCUSSION THREAD

**Discussion by DVoorhees1957** [GerryDazoo/Slinger#226] [General]
There appears to be an approach to add a new Slingbox to your Directory which doesn't rely on the "normal" web page (https://www.slingbox.com/getstarted), which appears to be broken/non-functional. The reason this matters is that, once the new device has been added to your Directory, you can then use the following URL to extract the admin password: https://newwatchsecure.slingbox.com/watch/slingAccounts/account_boxes_js

Basically, you log into your Sling account and go to https://accounts.sling.com/accounts/member/slingbox, then click on the "here" hyperlink at the end of the first sentence ("If you have a Pro-HD or older Slingbox you'd like to add to this account, you can do that here"). This opens up a "Register a new Slingbox" dialog box, which needs 3 things: the Slingbox ID (FinderID), the name you want to call the box, and a password. (The following sentence says, or at least implies, that this might not work for 350s or 500s, but I'm betting it would work even for those devices.)

The only caveat is that this approach depends critically on having the device's FinderID *without* the Slingbox already being registered to your account. I'm not sure that's possible, but I thought I'd raise the question - I'm sure there are people smarter than me who might know.

**Comment by GerryDazoo** [comment]
All slingboxes have finderids and their not hidden or at least not hard to find. If you don't configure a slingbox IP in my latest code it will scan the LAN and print out all the slingbox(es) IP port and FInderID.
Although I don't think you can change the 350/500 passwords. But give it a go and let us know.

**Reply by DVoorhees1957** [reply]
Cool, found my Slingbox 350's FinderID that way! Next step is to see if I can add it to my Sling Directory via the approach I outlined above...

**Reply by DVoorhees1957** [reply]
Darn. When I give it the FinderID and Name that your code found, I get the message that "This Slingbox has been already saved in your profile" - but it isn't there! (At least they don't show it in my Directory...)

**Reply by RandomMan719** [reply]
I just added an M1, M2 (from Ebay), and a pro to my account, yesterday or the day before.  For the M1 and M2, I just did a factory reset and configured them using SlingPlayer Desktop.  For the pro, I did a factory reset and set it up with Slingplayer 2.0.  All three are on my account now, so I just used the link:  https://newwatchsecure.slingbox.com/watch/slingAccounts/account_boxes_js  to get the passwords for the M1 and M2.  The pro allows you to create an admin password while configuring for the first time, or after factory reset.  They all show up in my directory.

**Reply by RandomMan719** [reply]
I have also done multiple factory resets on one of my ProHDs, to get the digital channel scan on Slingplayer 1.5.1, without logging into the servers, still no luck.  But I can do initial setup and digital scan using Slingplayer 2.0.

**Reply by gbjbasdw** [reply]
Question for RandomMan719 - you say you use Slingplayer 2.0 - is this for configuring a Slingbox using the Setup Assistant ? I can't get mine to works now (I'm using V1.5 for this purpose).  I get a dialogue box saying "This Adobe Flask ActiveX Installer is only intended for Windows 7 and prior. Please run the Adobe Flash Player ActiveX Installer for Windows 8 and Windows 10." I can't find anywhere to download this from as it seems to be discontinued. Can you shed any light ?

Steve

**Reply by flipper2222222** [reply]
> Question for RandomMan719 - you say you use Slingplayer 2.0 - is this for configuring a Slingbox using the Setup Assistant ? I can't get mine to works now (I'm using V1.5 for this purpose). I get a dialogue box saying "This Adobe Flask ActiveX Installer is only intended for Windows 7 and prior. Please run the Adobe Flash Player ActiveX Installer for Windows 8 and Windows 10." I can't find anywhere to download this from as it seems to be discontinued. Can you shed any light ?
> 
> Steve

I have a few flash files that you can have, they are not installers but you can copy them to the correct directories and register them by command line, note that this does present some security issues as flash as you say is discontinued

**Reply by sp222222** [reply]
> Question for RandomMan719 - you say you use Slingplayer 2.0 - is this for configuring a Slingbox using the Setup Assistant ? I can't get mine to works now (I'm using V1.5 for this purpose). I get a dialogue box saying "This Adobe Flask ActiveX Installer is only intended for Windows 7 and prior. Please run the Adobe Flash Player ActiveX Installer for Windows 8 and Windows 10." I can't find anywhere to download this from as it seems to be discontinued. Can you shed any light ?
> 
> Steve

Steve:

Check this recent thread https://github.com/GerryDazoo/Slinger/discussions/465

**Comment by DVoorhees1957** [comment]
OK, so here's a blow-by-blow of my (ultimately successful) attempt to register my old Slingbox 350, configure it (as though it was new), and finally extract the admin password. I'm writing these steps down because there are lots of ways for things to go wrong - and to highlight that sometimes you need to keep trying a step that seemingly failed.

The background: I purchased a Slingbox 350 in Jan 2015 and registered it with Sling. The following year, I bought a Slingbox 500, registered it with Sling, and "retired" the 350 to a closet (although I don't remember deleting the 350 from my Sling "Directory"). Anyway, some time after adding the 500 to my Directory in late 2016, the 350 disappeared from my Directory - which means I couldn't connect to it, and more importantly, it wouldn't show up in the results when I would access the web page that shows the admin password for registered devices (https://newwatchsecure.slingbox.com/watch/slingAccounts/account_boxes_js).

As a result, for all practical purposes I had to act as though I had just purchased the 350 from someone on eBay, which means I had to figure out how to add it to my Sling Directory.

The short version of the story is: I couldn't figure out any way to add it to my existing Sling account. The "advertised" way to do this, for a Slingbox 350 or 500, is to go to https://www.slingbox.com/getstarted - but that web page is broken/non-functional. If you enter that URL directly, you'll get an error message similar to the following: "The requested URL was rejected. Please consult with your administrator. Your support ID is: xxxxxxxxxxx". The other approach that has worked for some people is to use Slingplayer Desktop - but in my case, Slingplayer Desktop could only see my Slingbox 500, not my 350... and if I powered off my 500, Slingplayer Desktop couldn't see any devices - which only leaves you with the option to try to set up a new device, which takes you to the broken URL, yada yada... I even tried adding the 350 by using it's FinderID (which the Slinger code is gracious enough to display, if you give it a config.ini file that doesn't have any Slingboxes defined); see https://github.com/GerryDazoo/Slinger/discussions/226 for the results of that (futile) effort.

So, after taking a brief break (and consuming an alcoholic beverage or two), I came up with the idea of logging out of my existing Sling account, setting up a new account, and logging into that account. Voila! Slingplayer Desktop now saw both my 500 and my 350 (and both appeared to be already configured, judging by the little green circle to the left of both of them). I powered off my 500 so that it wouldn't cause problems (or get assigned to the new Sling account), then tried to use Slingplayer Desktop to set up the "new" 350. Alas, Slingplayer Desktop just sat and spun on the "Initializing" phase. So I tried to use the FinderID to register my "new" 350 per https://github.com/GerryDazoo/Slinger/discussions/226 ... sadly, I got the same error message when I tried this, "This Slingbox has been already saved in your profile". That might have been true when I was signed into my old, original Sling account, but was clearly **not** true for the new account. My belief at this point is that the "This Slingbox has been already saved in your profile" merely means that Sling already knows about this device and it's associated with *some* account, although not necessarily *your* account.

After another short break (and perhaps another beverage), I decided to try the hard reset route - just to make sure that the 350 got removed from Sling's database. I then tried Slingplayer Desktop, in desperation, one more time - and this time it showed just the 350 (since the 500 was still powered off), and showed it in "yellow" status, meaning that it needed to be set up/configured. I knew I was close!

Sadly, when I clicked the "Set up" button, Slingplayer Desktop got stuck in the "Initializing" phase... again!! I had to close Slingplayer Desktop and re-open it, several times, until I **finally** got to the Welcome screen.

At this point, I was hoping I was home free, and the rest of setup when pretty much as expected. Setup finished, and the "new" Slingbox 350 showed up in the Sling Directory of my (new) Sling account. Once that was done, I could finally extract the admin password via the URL https://newwatchsecure.slingbox.com/watch/slingAccounts/account_boxes_js.

I guess the moral of the story is: if at first you don't succeed, try try again. At least - and at last - I finally have the admin passwords for both of my Slingboxes!

**Reply by hexgoblin** [reply]
Thanks very much for this - going down the 'new account' route was the only way I was able to register a 350 I bought off eBay (by way of a backup) not long ago and extract the admin password.  Took me a few attempts at factory reset and several restarts of the Desktop app, but it worked (at 2am on the supposed shutdown day, no less).

**Reply by RandomMan719** [reply]
Nice, yeah I have been doing some factory resets on an M2 box.  It doesn't seem to take every time, but I have reset it a couple times and reconfigured it.  I am going to do it one last time and get the password, servers are still up. 8pm here.

**Comment by cupanther** [comment]
Where can you safely get a copy of the slingplayer desktop app?

**Reply by hexgoblin** [reply]
Don't know about today, but yesterday you could still get the latest version - which I used to setup my 350 -  from the newwatch.slingbox.com site.  Not sure about older versions.

**Comment by tom76017** [comment]
Interesting that the jason still works if you already have the Slingbox on your account.  https://newwatchsecure.slingbox.com/watch/slingAccounts/account_boxes_js

**Reply by cupanther** [reply]
I was able to hard reset on a 350 and set it up using the windows app today.

**Reply by RandomMan719** [reply]
Slingplayer Desktop on Windows10?

**Reply by cupanther** [reply]
Yes

**Comment by c12rider** [comment]
Is there anyway to get slinger to work with any model SB now that the servers are down? Trying to get a hard reset SB500 to work but stuck at it trying to connect to internet, slinger sees it, but when I try to start a stream I get an error code and a message kicking off other user then can’t kick off other user….

**Reply by bradut** [reply]
@c12rider It definitely is a way, as this is the purpose of this project.
[Not sure about any SB model: I think some SB models cannot be resuscitated after being bricked]

You may start with blocking the port the SB uses to phone home (5678), otherwise the device will get back an "update" which will brick it back.
Then, after a factory reseting, get your SB work with or without the Slinger server.

I used **this document** to get my SB work:
https://www.mediafire.com/file/ppw6ep09t99xhu0/Slingbox_ProHD_PC_Unbricking-Setup-Network-Guide.pdf/file
 
Initially I kept on using SlingPlayer 1.5 connected directly to the SlingBox, but then switched to Slinger + SlingerPlayer and VLC.
